commands/migrate_commands.go

```diff
@@ -194,64 +253,77 @@ var gotoCommand = cli.Command{
 
 		log.Infof("Migrating to version %d", toVersionInt)
 
-		currentVersion, err := migrate.Version(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		migrate, mctx, cancel := newMigrateWithCtx(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		defer cancel()
+		currentVersion, err := migrate.Version(mctx)
 		if err != nil {
 			logErr(err).Fatalf("failed to migrate to version %d", toVersionInt)
 		}
 
 		relativeNInt := toVersionInt - int(currentVersion)
 
-		pipe := pipep.New()
-		go migrate.Migrate(pipe, ctx.GlobalString("url"), ctx.GlobalString("path"), relativeNInt)
-		ok := readPipe(pipe)
-		if !ok {
-			os.Exit(1)
+		err = migrate.Migrate(mctx, relativeNInt)
+		if err != nil {
+			logErr(err).Fatalf("Failed to migrate to vefrsion %d", toVersionInt)
 		}
-		logCurrentVersion(ctx.GlobalString("url"), ctx.GlobalString("path"))
+		logCurrentVersion(mctx, migrate)
 		return nil
 	},
 }
 
-func logErr(err error) *log.Entry {
-	return log.WithError(err)
+func newMigrateWithCtx(url, migrationsPath string) (*migrate.Handle, context.Context, func()) {
+	done := make(chan struct{})
+	m, err := migrate.Open(url, migrationsPath, migrate.WithHooks(
+		func(f file.File) error {
+			log.Infof("Applying %s migration for version %d (%s)", f.Direction, f.Version, f.Name)
+			return nil
+		},
+		func(f file.File) error {
+			done <- struct{}{}
+			return nil
+		},
+	))
+	if err != nil {
+		log.Fatalf("Initialization failed: %s", err)
+	}
+	ctx, cancel := newOsInterruptCtx(done)
+	return m, ctx, cancel
 }
 
-// readPipe reads items from a chan and returns a boolean and the number of migration files executed
-func readPipe(pipe chan interface{}) (ok bool) {
-	okFlag := true
-	if pipe != nil {
-		for {
+// newOsInterruptCtx returns new context that will be cancelled
+// on os.Interrupt signal.
+func newOsInterruptCtx(done <-chan struct{}) (context.Context, func()) {
+	ctx, cancel := context.WithCancel(context.Background())
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
```

Comment:
A lib should never presume of context behaviour.
You must pass a `context` to migrate, and handle this stuff in https://github.com/db-journey/journey/ instead.
That way, you let the users chose where `context.Background()` is created.

Reply:
I supposed that `commands` package is not supposed to be used apart from journey CLI. And it doesn't presume context behavior. The only thing it "presume" is that context _might_ be canceled, which is kind of normal thing to presume when you work with context.
So passing context here from `journey` or whatever else won't change/break anything.

Reply:
If you talking about os signals processing - yes, it does makes sense to move it to journey's main package if `commands` is used elsewhere beyond journey.

Reply:
my bad, I mixed up with the `migrate` project.
You're right!